types/jest/index.d.ts

```diff
@@ -299,7 +325,7 @@ declare namespace jest {
 
     interface Each {
         // Exclusively arrays.
-        <T extends any[]>(cases: ReadonlyArray<T>): (name: string, fn: (...args: T) => any, timeout?: number) => void;
```

Comment:
Please keep this line before the newly inserted line.

Reply:
@ExE-Boss

The new line now covers this case tho?

Reply:
It&nbsp;will cause&nbsp;regressions to&nbsp;read‑write&nbsp;tuples with&nbsp;11&nbsp;or&nbsp;more distinct&nbsp;elements.

Reply:
I don't understand can you elaborate?

Reply:
This will break:

```ts
const cases: [
	/*  0: */ typeof String,
	/*  1: */ typeof Boolean,
	/*  2: */ typeof Symbol,
	/*  3: */ typeof BigInt,
	/*  4: */ typeof Object,
	/*  5: */ typeof Reflect,
	/*  6: */ typeof Math,
	/*  7: */ typeof Map,
	/*  8: */ typeof Set,
	/*  9: */ typeof WeakMap,
	/* 10: */ typeof WeakSet,
] = [
	/*  0: */ String,
	/*  1: */ Boolean,
	/*  2: */ Symbol,
	/*  3: */ BigInt,
	/*  4: */ Object,
	/*  5: */ Reflect,
	/*  6: */ Math,
	/*  7: */ Map,
	/*  8: */ Set,
	/*  9: */ WeakMap,
	/* 10: */ WeakSet,
];

it.each([cases])('doStuff', (...args) => {
	args; // $ExpectType [StringConstructor, BooleanConstructor, SymbolConstructor, BigIntConstructor, ObjectConstructor, typeof Reflect, Math, MapConstructor, SetConstructor, WeakMapConstructor, WeakSetConstructor]
});
```

Reply:
@ExE-Boss 
that is how it is currently behaving anyways? my implementation make the `args` to be exact for <=10 arguments.

Reply:
```
'fallback': Array<(T extends ReadonlyArray<infer U>? U: any)>
```

already cover what the following does.
```
 <T extends any[]>(cases: ReadonlyArray<T>): (name: string, fn: (...args: T) => any, timeout?: number) => void;
```

Reply:
No, that’s&nbsp;not&nbsp;what `Array<(T extends ReadonlyArray<infer U>? U: any)>` does.

The&nbsp;current type&nbsp;of&nbsp;`args` for&nbsp;very&nbsp;long&nbsp;tuples is&nbsp;the&nbsp;tuple, but&nbsp;for&nbsp;your&nbsp;code, it’s&nbsp;an&nbsp;array&nbsp;whose&nbsp;types is&nbsp;the&nbsp;union of&nbsp;all&nbsp;types present&nbsp;in&nbsp;the&nbsp;tuple.

Reply:
@ExE-Boss

Thank you for the thorough review. But I think this line will actually infer `args` to be array of union of types.

See:
http://www.typescriptlang.org/play/#code/GYVwdgxgLglg9mABAUwIYQBYB4AqKAeUyYAJgM6KpgCeA2gLoB8AFBKmcmQFyIBKaJBABtqAQQBO41NVyMAlD2bAwigHTrU4gObdEOOYgC8jSjQPHEANzgwSiAN4AoRC8TjkUEOKRKw5k07OrgC+jqGOaJjMtLQAjAA0iABMiQDkqKn09HLMzPiJ1IkAXv4OQS5hckA

Reply:
In&nbsp;your&nbsp;example, you’re&nbsp;actually&nbsp;supplying `(string | number)[]`, but&nbsp;in&nbsp;my&nbsp;case, it&nbsp;will&nbsp;be&nbsp;inferred&nbsp;correctly:&nbsp;[🔗](https://www.typescriptlang.org/play/?ssl=10&ssc=16&pln=10&pc=17#code/GYVwdgxgLglg9mABAUwIYQBYB4AqKAeUyYAJgM6KpgCeA2gLoB8AFBKmcmQFyIBKaJBABtqAQQBO41NVyMAlD2bAwigHTrU4gObdEOOYgC8jSjQPHEANzgwSiAN4AoRC8TjkUEOKRKw5k07OrgC+jqGOaJjMtLQAjAA0iABMiQDkqKn0lBRxiSmI6Zn0cszM+InUiQBe-g5BLhAIZFCI+Ek8sUatANz1iI1gzYjU7cld1L2u-U0tVaOFXVW9wXJAA)

Reply:
ahhhh. make sense now.

Sorry for being stubborn on this and will put a fix in now.

Reply:
@ExE-Boss 

Can you think of a better way to do the fallback?